includes/User.php

```diff
@@ -4218,6 +4218,20 @@ public static function crypt( $password, $salt = false ) {
 	 * @return Boolean
 	 */
 	public static function comparePasswords( $hash, $password, $userId = false ) {
+		// Wikia change - begin
+		// @see PLATFORM-2502 comparing new passwords in PHP code.
+		// @todo mech remove after the new password hashing is implemented.
+		$caller = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS)[1];
+		Wikia\Logger\WikiaLogger::instance()->debug(
+			'NEW_HASHING comparePasswords called in PHP',
+			[
+				'user_id' => $userId,
+				'caller' => $caller['class'].'::'.$caller['function'],
+				'exception' => new Exception()
```

Comment:
To log the stacktrace, so it will be easier to see where the call came from and fix it to use Helios. I had an option of using Exception handling already supported by WikiaLogger or formatting the result of debug_backtrace myself. Also, I think we use this approach in couple of other places.